example/provider-extensions/quic-relay/quic-relay-client.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: relay
+  labels:
+    app: gardener-api-quic-client
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: gardener-api-quic-client
+  namespace: relay
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: gardener-api-quic-client
+  template:
+    metadata:
+      labels:
+        app: gardener-api-quic-client
+    spec:
+      containers:
+      - args:
+        - --server=$HOST:443
+        - --upstream=gardener-apiserver.relay.svc.cluster.local:443
+        - --cert-file=/certs/tls.crt
+        - --cert-key=/certs/tls.key 
+        - --ca-file=/certs/ca.crt
+        - --v=2
+        image: ghcr.io/mvladev/quic-reverse-http-tunnel/quic-client-tcp:v0.1.3
```

Comment:
An SSH tunnel could be used too. This would not be very complicated I guess and we will find many well maintained images which include ssh.
I just found the quic-tunnel configuration at a different point in `gardener/gardener` and reused the scripts.